.github/scripts/test_label_utils.py

```diff
@@ -0,0 +1,52 @@
+from typing import Any
+
+from unittest import TestCase, mock, main
+from label_utils import (
+    get_last_page_num_from_header,
+    gh_get_labels,
+)
+
+
+class TestLabelUtils(TestCase):
+    MOCK_HEADER_LINKS_TO_PAGE_NUMS = {
+        1: {"link": "<https://api.github.com/dummy/labels?per_page=10&page=1>; rel='last'"},
+        2: {"link": "<https://api.github.com/dummy/labels?per_page=1&page=2>;"},
+        3: {"link": "<https://api.github.com/dummy/labels?per_page=1&page=2&page=3>;"},
+    }
+
+    def test_get_last_page_num_from_header(self) -> None:
+        for expected_page_num, mock_header in self.MOCK_HEADER_LINKS_TO_PAGE_NUMS.items():
+            self.assertEqual(get_last_page_num_from_header(mock_header), expected_page_num)
+
+    @mock.patch("label_utils.get_last_page_num_from_header", return_value=3)
+    @mock.patch("label_utils.request_for_labels", return_value=("foo", "bar"))
+    @mock.patch("label_utils.update_labels")
+    def test_gh_get_labels(
+        self,
+        mock_update_labels: Any,
+        mock_request_for_labels: Any,
+        mock_get_last_page_num_from_header: Any,
+    ) -> None:
+        gh_get_labels("foo", "bar")
+        mock_get_last_page_num_from_header.assert_called_once()
+        self.assertEqual(mock_update_labels.call_count, 3)
+        self.assertEqual(mock_request_for_labels.call_count, 3)
+
+    @mock.patch("label_utils.get_last_page_num_from_header", return_value=0)
+    @mock.patch("label_utils.request_for_labels", return_value=("foo", "bar"))
+    @mock.patch("label_utils.update_labels")
+    def test_gh_get_labels_raises_with_no_pages(
+        self,
+        mock_update_labels: Any,
+        mock_request_for_labels: Any,
+        get_last_page_num_from_header: Any,
+    ) -> None:
+        with self.assertRaises(AssertionError):
+            gh_get_labels("foo", "bar")
+            get_last_page_num_from_header.assert_called_once()
+            mock_update_labels.assert_called_once()
```

Comment:
Both of these asserts also seem overly specific. The only thing you're really caring about here is if the `gh_get_labels` throws an assertion, and ideally you'd check to see if the assertion contains a specific substring (e.g. "number of pages")